modules/loaders-common/src/main/scala/com/snowplowanalytics/snowplow/loaders/transform/NonAtomicFields.scala

```diff
@@ -61,7 +62,7 @@ object NonAtomicFields {
         // Remove whole schema family if there is no subversion left after filtering
         subVersions.nonEmpty
       }
-      .traverse { case (tabledEntity, subVersions) =>
+      .parTraverse { case (tabledEntity, subVersions) =>
```

Comment:
This will generate a bunch of requests to the Iglu server at once. But I think that this is fine as each instance should be requesting only a few schemas at once, and even if we have tens of instances, it is very unlikely that all of them resolve the schemas at the exact same time.

Reply:
That is a very good point.  It could be many 10s of schemas.  I think we get best separation of concerns if stuff like that to be controlled by the connection pool in the HTTP client.  Blaze by default allows 256 concurrent connections per server, and that is probably a bit too large for us.

I might make this change in combination with adding a default common-streams HTTP client where we override some config options like max connections per server.

Reply:
> It could be many 10s of schemas

If we have only one loader requesting that that's fine (e.g. one collector instance can handle constantly more than 1000 requests / second). If we have tens of loaders requesting it, that's a little bit scarier, but that's still probably fine, it's unlikely to happen, and there is your new `exitOnMissingSchema` feature (which I think is a great idea).

>  I think we get best separation of concerns if stuff like that to be controlled by the connection pool in the HTTP client.

I very much agree ! This code shouldn't worry about the under-the-hood HTTP client

> I might make this change in combination with adding a default common-streams HTTP client where we override some config options like max connections per server

:ok_hand:

Reply:
Opened #87 to address the http client connection pool.